AntennaTracker/mode_guided.cpp

```diff
@@ -3,15 +3,23 @@
 #include "Tracker.h"
 #include <GCS_MAVLink/GCS.h>
 
+constexpr uint32_t DEBUG_TIMEOUT = 5000;
+constexpr float eps = 0.01f;
+
 void ModeGuided::update()
 {
+    static float last_target_yaw = NAN, last_target_pitch = NAN;
     static uint32_t last_debug;
     const uint32_t now = AP_HAL::millis();
     float target_roll, target_pitch, target_yaw;
     _target_att.to_euler(target_roll, target_pitch, target_yaw);
-    if (now - last_debug > 5000) {
+    bool yaw_changed = isnan(last_target_yaw) || fabsf(last_target_yaw - target_yaw) > eps;
+    bool pitch_changed = isnan(last_target_pitch) || fabsf(last_target_pitch - target_pitch) > eps;
+    if (now - last_debug > DEBUG_TIMEOUT && (yaw_changed || pitch_changed)) {
         last_debug = now;
         gcs().send_text(MAV_SEVERITY_INFO, "target_yaw=%f target_pitch=%f", degrees(target_yaw), degrees(target_pitch));
+        last_target_yaw = target_yaw;
+        last_target_pitch = target_pitch;
```

Comment:
Not really a fan off this. We try not to use statics and nans.

How about adding a "last_set_angle_ms" timestamp for the "set_angle" call which updates the `_target_att`. Then you could check if the timestamp is larger than the last debug print.